Error: "Strategic_Movement.cc: Failed to add movement event." in tactical screen

[//]: # (Thank you for taking the time to report an issue or propose an enhancement. Questions are also welcome.)
[//]: # (If you have more than one item to report, please open separate issues!)

[//]: # (This text will not be visible and you can freely delete it. Check the Preview tab for a cleaner look.)

### What is wrong
Error on the tactical screen while in combat, but seems caused by a strategic game event. 

About 1 minute from Savegame time, at 21:10 ingame time, the error occurs. The error is independent of what I do in tactical (can just idle until error). 
In v0.19.1 this crashed the game. In v0.20.0 the game stays alive but the error still shows in the log. 

I am in combat in H14 Alma. At the same time I have another squat (Squat 2, Hamous and Bull) retreat from C9 to D9. Both C9 and D9 are enemy sectors. The ice cream truck is left in C9. There are no other player squats moving. 

I have tried to reverse direction of squat 2 back into C9, but that has no effect and the game errors regardless.

| JA2S version | Game version        | Vanilla bug? | Used to work? |
| ------------ | ------------------- | ------------ | ------------- |
|  v0.20.0-git+439a6d3 | eg. JA2 Gold steam  |    ?    |    yes     |

This is played on Straciatella without any mini-mods loaded.

To reproduce: Open saved game, wait until error shows in the log (at 21:10 ingame time).

```
2022-11-14T10:01:43 [INFO] game\GameLoop.cc: Version Label: Stracciatella v0.20.0-git+439a6d3
2022-11-14T10:01:43 [INFO] game\GameLoop.cc: Version #:     Build 04.12.02
2022-11-14T10:01:50 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:50 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:50 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:50 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:50 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:50 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:50 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:50 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:50 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:51 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:51 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:51 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:51 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:51 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:51 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:51 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:51 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:51 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:51 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:51 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:52 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:52 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:01:52 [WARN] game\Strategic\StrategicMap_Secrets.cc: No secret defined at sector 62
2022-11-14T10:03:02 [ERROR] game\Strategic\Strategic_Movement.cc: Failed to add movement event.
```
[2022-11-13t16-40-32z-7-Error failed to add movement event.zip](https://github.com/ja2-stracciatella/ja2-stracciatella/files/10001877/2022-11-13t16-40-32z-7-Error.failed.to.add.movement.event.zip)

--- a/src/game/Strategic/Strategic_Movement.cc
+++ b/src/game/Strategic/Strategic_Movement.cc
@@ -1081,7 +1081,8 @@ void GroupArrivedAtSector(GROUP& g, BOOLEAN const check_for_battle, BOOLEAN cons
 		if (g.fVehicle)
 		{
 			VEHICLETYPE const& v = GetVehicleFromMvtGroup(g);
-			if (!IsHelicopter(v) && !g.pPlayerList)
+			// Group size of 1 means the vehicle is the only member of this group.
+			if (!IsHelicopter(v) && g.ubGroupSize <= 1)
 			{ /* Nobody here, better just get out now. With vehicles, arriving empty
 				* is probably ok, since passengers might have been killed but vehicle
 				* lived. */
@@ -1120,18 +1121,13 @@ void GroupArrivedAtSector(GROUP& g, BOOLEAN const check_for_battle, BOOLEAN cons
 	{ /* Queue battle! Delay arrival by a random value ranging from 3-5 minutes,
 		* so it doesn't get the player too suspicious after it happens to him a few
 		* times, which, by the way, is a rare occurrence. */
-		if (AreInMeanwhile())
-		{ /* Tack on only 1 minute if we are in a meanwhile scene. This effectively
-			* prevents any battle from occurring while inside a meanwhile scene. */
-			++g.uiArrivalTime;
-		}
-		else
-		{
-			g.uiArrivalTime += Random(3) + 3;
-		}
+
+		/* Tack on only 1 minute if we are in a meanwhile scene. This effectively
+		 * prevents any battle from occurring while inside a meanwhile scene. */
+		g.setArrivalTime(g.uiArrivalTime + (AreInMeanwhile() ? 1 : (Random(3) + 3)));
 
 		if (!AddStrategicEvent(EVENT_GROUP_ARRIVAL, g.uiArrivalTime, g.ubGroupID))
-			SLOGA("Failed to add movement event.");
+			SLOGA("GroupArrivedAtSector: Failed to add movement event.");
 
 		if (g.fPlayer && g.uiArrivalTime - ABOUT_TO_ARRIVE_DELAY > GetWorldTotalMin())
 		{
@@ -1627,7 +1623,7 @@ static void DelayEnemyGroupsIfPathsCross(GROUP& player_group)
 		 * values to figure out how far along its route a group is! */
 		g.setArrivalTime(player_group.uiArrivalTime + 1 + Random(10));
 		if (!AddStrategicEvent(EVENT_GROUP_ARRIVAL, g.uiArrivalTime, g.ubGroupID))
-			SLOGA("Failed to add movement event.");
+			SLOGA("DelayEnemyGroupsIfPathsCross: Failed to add movement event.");
 	}
 }
 
@@ -1770,7 +1766,7 @@ static void InitiateGroupMovementToNextSector(GROUP* pGroup)
 
 	//Post the event!
 	if( !AddStrategicEvent( EVENT_GROUP_ARRIVAL, pGroup->uiArrivalTime, pGroup->ubGroupID ) )
-		SLOGA("Failed to add movement event.");
+		SLOGA("InitiateGroupMovementToNextSector: Failed to add movement event.");
 
 	//For the case of player groups, we need to update the information of the soldiers.
 	if( pGroup->fPlayer )
@@ -2762,7 +2758,7 @@ void RetreatGroupToPreviousSector(GROUP& g)
 	}
 
 	if (!AddStrategicEvent(EVENT_GROUP_ARRIVAL, g.uiArrivalTime, g.ubGroupID))
-		SLOGA("Failed to add movement event.");
+		SLOGA("RetreatGroupToPreviousSector: Failed to add movement event.");
 
 	// For the case of player groups, we need to update the information of the soldiers.
 	if (g.fPlayer)

